src/STW.ProcessingApi.Function.UnitTests/Services/BcpServiceTest.cs

```diff
@@ -0,0 +1,77 @@
+namespace STW.ProcessingApi.Function.UnitTests.Services;
+
+using System.Net;
+using FluentAssertions;
+using Function.Services;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Models;
+using Moq;
+using TestExtensions;
+
+[TestClass]
+public class BcpServiceTest
+{
+    [TestMethod]
+    public async Task GetBcpsWithCodeAndType_ReturnsBcps_WhenValues()
+    {
+        // Arrange
+        var bcp = new Bcp
+        {
+            Id = 1,
+            Code = "CODE",
+            Name = "Name",
+            Suspended = false,
+        };
+
+        var httpMessageHandler = new Mock<HttpMessageHandler>();
+        httpMessageHandler
+            .SetupSendAsync(HttpMethod.Get, "https://bcp.service/bcps/search?code=CODE&type=CHED_TYPE")
+            .ReturnsHttpResponseAsync(
+                new BcpSearchResponse
+                {
+                    Bcps = [bcp],
+                },
+                HttpStatusCode.OK);
+
+        var httpClient = new HttpClient(httpMessageHandler.Object)
+        {
+            BaseAddress = new Uri("https://bcp.service"),
```

Comment:
Introduce a [TestInitialize] annotation and method to this class and instantiate your system under test, HTTP client and mocks there rather than duplicate them.

Reply:
They change in each test

Reply:
The mock instantiation and HTTP Client don't change? Please take a look at this test class:
https://github.com/DEFRA/stw-processing-api/blob/a0251a95fa2eb1d7c4b0684c75e201d1bfbb6261/src/STW.ProcessingApi.Function.UnitTests/Services/ApprovedEstablishmentServiceTests.cs